### PR TITLE
Verification cephadm shell when running from a non-monitor node

### PR DESCRIPTION
**Pull request by MohitBis**
Automated : Verification cephadm shell when running from a non-monitor node
Test case covered : CEPH-83575009
Polarion Link : https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83575009
Signed-off-by: Mohit <mobisht@redhat.com>

# Description
Automated : Verification cephadm shell when running from a non-monitor node
Test case covered : CEPH-83575009
Test Results: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-4PCNYA

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [✓] Create a test case in Polarion reviewed and approved.
- [✓] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [✓] Review the automation design
- [✓] Implement the test script and perform test runs
- [✓] Submit PR for code review and approve
- [✓] Update Polarion Test with Automation script details and update automation fields
- [✓] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/cephadm/test_cephadm_shell.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            cmd="mkdir -p /var/lib/ceph && touch /var/lib/ceph/ceph-test", sudo=True
+        )
+        return 0
+    except Exception as e:
+        log.error(e)
+        return 1
+
+
+def run(ceph_cluster, **kw):
+    """Verification cephadm shell when running from a non-monitor node
```
Comment (Manohar-Murthy) [inline tests/cephadm/test_cephadm_shell.py:27]: HI Mohit, how are we selecting non-monitor node?

Comment (MohitBis) [reply tests/cephadm/test_cephadm_shell.py:27]: Thanks Manohar

Below set of method select "installer node"(non-monitor node).

L45 and L47

instance = CephAdmin(cluster=ceph_cluster, **config)
log.info("Creating a mon dir and daemon specfic file")
create_mon_daemon_file(instance.installer)

File: suites/pacific/upgrades/tier-1_upgrade_test-4x-to-5x-container.yaml [hunk 2]
```diff
@@ -157,3 +157,8 @@ tests:
         config-file-name: test_multitenant_access.yaml
         timeout: 300
     desc: Perform rgw tests
+
+- test:
+    name: test_cephadm_shell.py
```
Comment (sunilangadi2) [inline suites/pacific/upgrades/tier-1_upgrade_test-4x-to-5x-container.yaml:162]: @MohitBis Provide the proper test name not the module name. add polarion id  as well here.
Provide PR link in polarion automation script field and update polarion automation field.

**Review by sunilangadi2 — CHANGES_REQUESTED** [review]

File: suites/pacific/upgrades/tier-1_upgrade_test-4x-to-5x-container.yaml (hunk 2 above)
Comment (MohitBis) [reply suites/pacific/upgrades/tier-1_upgrade_test-4x-to-5x-container.yaml]: Thanks Sunil.

As per the comment.
1. Changed test file name to  "test_cephadm_shell_run.py".
2. Update PR in polarion same added in description.

File: tests/cephadm/test_cephadm_shell.py (hunk 1 above)
Comment (Manohar-Murthy) [reply tests/cephadm/test_cephadm_shell.py]: Thanks.

**Review by Manohar-Murthy — DISMISSED** [review]
looks good.

File: suites/pacific/upgrades/tier-1_upgrade_test-4x-to-5x-container.yaml [hunk 3]
```diff
@@ -157,3 +157,8 @@ tests:
         config-file-name: test_multitenant_access.yaml
         timeout: 300
     desc: Perform rgw tests
+
+- test:
+    name: test_cephadm_shell_run.py
```
Comment (Ameenasuhani) [inline suites/pacific/upgrades/tier-1_upgrade_test-4x-to-5x-container.yaml:162]: Add polarion id to the suite

**polarion-id: CEPH-83575009**

**Review by Ameenasuhani — CHANGES_REQUESTED** [review]

File: suites/pacific/upgrades/tier-1_upgrade_test-4x-to-5x-container.yaml (hunk 2 above)
Comment (MohitBis) [reply suites/pacific/upgrades/tier-1_upgrade_test-4x-to-5x-container.yaml]: @sunilangadi2 
As per the discussion.
1. Added new method(getting_non_mon_node) to support non-mon node selection.
2. Updated polarion id.

File: suites/pacific/upgrades/tier-1_upgrade_test-4x-to-5x-container.yaml (hunk 3 above)
Comment (MohitBis) [reply suites/pacific/upgrades/tier-1_upgrade_test-4x-to-5x-container.yaml]: Thanks @Ameenasuhani 
Updated

File: suites/pacific/upgrades/tier-1_upgrade_test-4x-to-5x-container.yaml (hunk 2 above)
Comment (sunilangadi2) [reply suites/pacific/upgrades/tier-1_upgrade_test-4x-to-5x-container.yaml]: Thanks, @MohitBis Changes LGTM
Can you please add test run logs here with new changes?

Comment (MohitBis) [reply suites/pacific/upgrades/tier-1_upgrade_test-4x-to-5x-container.yaml]: Thanks @sunilangadi2 
Please find below test run logs for new changes
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-GBK7R5

**Review by sunilangadi2 — APPROVED** [review]

**Review by Ameenasuhani — APPROVED** [review]
LGTM

**Review by vdas-redhat — APPROVED** [review]
Good to merge.

**Comment by manasagowri** [comment]
@MohitBis This test case when integrated with the complete upgrade test suite has failed. PFB the failure logs.
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-ZGSNZS/test_cephadm_shell_run_0.log 
Could you please recheck and raise a PR if necessary.
Also from next time please make sure to perform an end to end test and attach logs before merging the PR.

cc: @vdas-redhat @Manohar-Murthy

**Comment by MohitBis** [comment]
Hi @manasagowri 
Failure is expected.

Since conf file used for run not have any mon node, TC is expected test cephadm shell in non mon node.

In the test suit "tier-1_upgrade_test-4x-to-5x-container.yaml" it is mentioned to use conf file "conf/pacific/upgrades/upgrades.yaml" which is not used in above run.

https://github.com/red-hat-storage/cephci/blob/master/suites/pacific/upgrades/tier-1_upgrade_test-4x-to-5x-container.yaml#L4

As discussed I will add string "Particular test suite needs at least one non mon node to exist for that test to work" in suite and raise separate PR.

cc: @vdas-redhat @Manohar-Murthy

**Pull request merged**